src/07-even-more-loops.js

```diff
@@ -44,7 +63,26 @@ function shoutWhileLoop(array) {}
  *  shoutForOfLoop([ "A", "Very", "Happy", "Array" ]);
  *  //> [ "A!", "Very!", "Happy!", "Array!" ];
  */
-function shoutForOfLoop(array) {}
+function shoutForOfLoop(array) {
+  {
+    let newArr = []
+    let i = 0
+    for (let element of array) {
+      array[i] = array[i] + '!'
+      i++
+    }
+    return array
+  }
+}
+//Brainstorming & Notes ---V
+//claims its not a for loop in the npm test result->needs a for...of loop
+// for (let value of i = 0) {
+//   i < array.length
+
+//   newArr.push(array[i] + '!')
+//   i++;
+//}
+//element is i in array, what we are looking through the array for an element
```

Comment:
This works,  but you're using `i` when you don't have to. This is the beauty of the `for of` loop. Try solving this one without `i`!